tests/functional/test_broken_stdout.py

```diff
@@ -47,13 +45,16 @@ def test_broken_stdout_pipe(deprecated_python: bool) -> None:
     assert returncode == _BROKEN_STDOUT_RETURN_CODE
 
 
-def test_broken_stdout_pipe__log_option(deprecated_python: bool, tmpdir: Path) -> None:
+def test_broken_stdout_pipe__log_option(
+    deprecated_python: bool,
+    tmp_path: pathlib.Path,
+) -> None:
     """
     Test a broken pipe to stdout when --log is passed.
     """
-    log_path = os.path.join(str(tmpdir), "log.txt")
+    log_path = tmp_path.joinpath("log.txt")
```

Comment:
Any particular reason to use `joinpath` here, rather than `tmp_path / "log.txt"`? I prefer the operator personally, but it's not a big deal. It might be something we want a style guideline on which to use, though, to avoid arguments 😉